models/app.models.js

```diff
@@ -104,3 +104,23 @@ exports.selectArticleToPatch = (article_id, inc_votes) => {
     }
   });
 };
+
+exports.selectCommentToDelete = (comment_id) =>{
+  let queryString = ``
+  const queryValue = []
+  const queryProms = []
+  if (comment_id){
+    queryString += `DELETE FROM comments WHERE comment_id = $1 RETURNING*`
+    queryValue.push(comment_id)
+    queryProms.push(checkExists("comments", "comment_id", comment_id))
+  }
+  queryProms.push(db.query(queryString, queryValue))
+  return Promise.all(queryProms)
+  .then((promResults)=>{
+    if (queryProms.length === 1) {
+      return promResults[0].rows[0];
```

Comment:
I don't think we will ever hit this if statement, which asks the question should it be there, if our checkExists fails then it should reject a promise and we caught by our controller.